packages/manager-components/src/hooks/iam/useOvhIam.tsx

```diff
@@ -0,0 +1,116 @@
+import { useState, useEffect } from 'react';
+import { apiClient } from '@ovh-ux/manager-core-api';
+import {
+  useInfiniteQuery,
+  useQuery,
+  keepPreviousData,
+} from '@tanstack/react-query';
+import { IamInterface, PageInterface, PageReduceIam } from './iam.interface';
+
+export const fetchAuthorizationsCheck = async ({
+  actions,
+  urns,
+}: {
+  actions: string[];
+  urns: string[];
+}) => {
+  const { data } = await apiClient.v2.post(`/iam/authorization/check`, {
+    actions,
+    resourceURNs: urns,
+  });
+  return data;
+};
+
+function reducePages(pages: PageInterface[]) {
+  const mergedPages = pages.reduce(
+    (acc: PageReduceIam, page: PageInterface) => {
+      if (acc[page.resourceURN]) {
+        acc[page.resourceURN].authorizedActions = [
+          ...new Set([
+            ...acc[page.resourceURN].authorizedActions,
+            ...page.authorizedActions,
+          ]),
+        ];
+        acc[page.resourceURN].unauthorizedActions = [
+          ...new Set([
+            ...acc[page.resourceURN].unauthorizedActions,
+            ...page.unauthorizedActions,
+          ]),
+        ];
+      } else {
+        acc[page.resourceURN] = {
+          ...page,
+          authorizedActions: [...new Set(page.authorizedActions)],
+          unauthorizedActions: [...new Set(page.unauthorizedActions)],
+        };
+      }
+      return acc;
+    },
+    {},
+  );
+  const uniquePages = Object.values(mergedPages);
+  return uniquePages;
+}
+export function useInfiniteAuthorizationsIam({
+  actions = [],
+  urns = [],
+  isTrigger = true,
+  pageIndex = 0,
+}: IamInterface) {
+  const [authorizedActions, setAuthorizedActions] = useState([]);
+  const [previousPageIndices, setPreviousPageIndices] = useState([]);
+
+  const { data, fetchNextPage } = useInfiniteQuery({
+    queryKey: ['iam-authorization-infinite'],
```

Comment:
**nitpick:** As @frenautvh suggested, this has to be combination of `actions` and `urns` to make it unique.